satpy/composites/__init__.py

```diff
@@ -1647,35 +1669,81 @@ def __call__(self, projectables, *args, **kwargs):
         if len(projectables) != 2:
             raise ValueError("Expected 2 datasets, got %d" % (len(projectables),))
         projectables = self.match_data_arrays(projectables)
-        cloud_mask = projectables[1]
-        cloud_mask_data = cloud_mask.data
-        data = projectables[0]
-        alpha_attrs = data.attrs.copy()
-        if 'bands' in data.dims:
-            data = [data.sel(bands=b) for b in data['bands'] if b != 'A']
+        data_in = projectables[0]
+        mask_in = projectables[1]
+        mask_data = mask_in.data
+
+        alpha_attrs = data_in.attrs.copy()
+        if 'bands' in data_in.dims:
+            data = [data_in.sel(bands=b) for b in data_in['bands'] if b != 'A']
         else:
-            data = [data]
+            data = [data_in]
 
         # Create alpha band
         alpha = da.ones((data[0].sizes['y'],
                          data[0].sizes['x']),
                         chunks=data[0].chunks)
 
-        # Modify alpha based on transparency per class from yaml
-        flag_meanings = cloud_mask.attrs['flag_meanings']
-        flag_values = cloud_mask.attrs['flag_values']
+        for condition in self.conditions:
+            method = condition['method']
+            value = condition.get('value', None)
+            if isinstance(value, str):
+                value = _get_flag_value(mask_in, value)
+            transparency = condition['transparency']
+            mask = self._get_mask(method, value, mask_data)
 
-        if isinstance(flag_meanings, str):
-            flag_meanings = flag_meanings.split()
+            if transparency == 100.0:
+                data = self._set_data_nans(data, mask, alpha_attrs)
+            alpha_val = 1. - transparency / 100.
+            alpha = da.where(mask, alpha_val, alpha)
 
-        for key, val in self.transparency.items():
-            if isinstance(key, str):
-                key_index = flag_meanings.index(key)
-                key = flag_values[key_index]
-            alpha_val = 1. - val / 100.
-            alpha = da.where(cloud_mask_data == key, alpha_val, alpha)
         alpha = xr.DataArray(data=alpha, attrs=alpha_attrs,
                              dims=data[0].dims, coords=data[0].coords)
         data.append(alpha)
+
         res = super(MaskingCompositor, self).__call__(data, **kwargs)
         return res
+
+    def _get_mask(self, method, value, mask_data):
+        """Get mask array from *mask_data* using *method* and threshold *value*.
+
+        The *method* is the name of a numpy function.
+
+        """
+        try:
+            func = getattr(np, method)
```

Comment:
So in principle this means any numpy method can be called. Just for security reasons, would you mind adding a check that `method` is not outside what we expect ?

Reply:
I added a list of accepted methods, and an unit test for trying to use something else. I hope I listed all the currently relevant methods.